### DISCUSSION TITLE
How to learn programming？

### DISCUSSION THREAD

**Discussion by cxman1** [community/community#36424] [Other Feature Feedback, Questions, & Ideas] [answered]
### Programming is really brain burning. It's hard to make achievements
I need the help of experts 😵😶‍🌫️😵

**Comment by HadrienBoyer** [comment]
You should start to read and check on websites that gives you tutorials, read code and practice! There is a lot of way to start, you can read a book for the start and to learn the basics, and then start the tutorials/exercices that are largely available on the web, including YouTube (I like YouTube's tutorials because it's free and the "teacher" usually know his topic and explain each line of code).

Have a nice journey into development! :)

**Reply by cxman1** [reply]
Thank you for your answer. Thank you very much

**Comment by jfullstackdev** [comment, marked as answer]
and it's really frustrating most of the time, even if you are already a software developer, if you watch even, say, some Tech Leads in top-tech companies, they are saying that

you will feel you are not progressing too, and you can't feel your work is done, that's the nature of being a developer, that's why, other IT companies are applying the Agile way, so at least, the team can feel a progress

that's same thing when I was hired as a developer, I never had this feeling for 8 years as a teacher, 

and this was me when I was hired, during the times when I was learning programming, that was so frustrating even if, say, you already learned one programming language then you go to next, particularly, say, from a BASIC language to an OOP, they are really very different when it comes to syntax, and say, from a language to a framework, that will be always a daunting task as if you don't know anything

so how did I progress ? first, I had the interest of programming computer, the concept of programmable computer, that you can program, say, a business problem solution or 3D games, 

I'd never known that in my childhood, that computer games were actually being programmed, in my mind, that's the computer that created these games automatically,

so you need to write programs, even small ones, out of your interest, just pick one language, I started in PureBasic, you can pick Python

you just need to learn the core, that's actually very important, most of your coding problems will revolve around that: variables, data types, arrays, conditionals, loops etc., 

here is my own list

https://github.com/jdevstatic/programming-core-concepts

and as you progress, there will be interesting problems that you will be needing to solve, for example, I never really had the interest of learning JavaScript, but when I was creating my static site, I really need to use it and CSS, so, as I progressed I'm learning that at the same time, take note, even in real-world programming, you need to keep that habit of learning, it's always happening

**Reply by cxman1** [reply]
In the process of learning, many knowledge points are too detailed and tedious. After learning new knowledge for a period of time, I will always forget some details of knowledge. The chain reaction of programming leads to the collapse of the whole program. I need to settle down well.

**Reply by jfullstackdev** [reply]
just small programs will do, along the way, you will learn how to decouple components, in such a way that when you add or change one functionality, side effects should be minimum but again, it's in real scenario, just practice grouping task into functions at first, make sure it only do one task and if it's becoming too long, you might want to break it to several functions too, variables that are general in scope should be at least at minimum

speaking of forgetting details, yup, I can't even write code without documentation, it's always necessary, you don't need to memorize all of the syntax

**Reply by cxman1** [reply]
Yes, thank you for your answer. I will go to read your article. At present, there will be jet lag between our side and your place. It's 12 o'clock in the morning here. Thank you very much for your generous answer.

**Comment by ossanna16** [comment]
Great question @cxman1, I'm happy you found our community 🙂
